kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/config/VirtualCluster.java

```diff
@@ -5,20 +5,80 @@
 public record VirtualCluster(TargetCluster targetCluster,
-                             @JsonProperty(required = true) ClusterNetworkAddressConfigProviderDefinition clusterNetworkAddressConfigProvider,
-                             @JsonProperty() Optional<Tls> tls,
+                             @Deprecated(forRemoval = true, since = "0.11.0") ClusterNetworkAddressConfigProviderDefinition clusterNetworkAddressConfigProvider,
+                             @Deprecated(forRemoval = true, since = "0.11.0") @JsonProperty() Optional<Tls> tls,
+
+                             @JsonProperty(required = false) List<VirtualClusterListener> listeners,
                              boolean logNetwork,
                              boolean logFrames,
                              @Nullable List<String> filters) {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(VirtualCluster.class);
+
+    /**
+     * Name given to the listener defined using the deprecated fields.
+     */
+    @Deprecated(forRemoval = true, since = "0.11.0")
+    static final String DEFAULT_LISTENER_NAME = "default";
+
+    @SuppressWarnings({ "removal", "java:S2789" }) // S2789 - checking for null tls is the intent
+    public VirtualCluster {
+        if (clusterNetworkAddressConfigProvider != null || (tls != null && tls.isPresent())) {
+            if (clusterNetworkAddressConfigProvider == null) {
+                throw new IllegalConfigurationException("Deprecated virtualCluster property 'tls' supplied, but 'clusterNetworkAddressConfigProvider' is null");
+            }
+            if (listeners == null || listeners.isEmpty()) {
+                LOGGER.warn(
+                        "The virtualCluster properties 'clusterNetworkAddressConfigProvider' and 'tls' are deprecated, specify virtual cluster listeners using the listeners map.");
+                listeners = List.of(new VirtualClusterListener(DEFAULT_LISTENER_NAME, clusterNetworkAddressConfigProvider, tls));
+            }
+            else {
+                throw new IllegalConfigurationException(
+                        "When using listeners, the virtualCluster properties 'clusterNetworkAddressConfigProvider' and 'tls' must be omitted");
+            }
+        }
+        if (listeners == null || listeners.isEmpty()) {
+            throw new IllegalConfigurationException("no listeners configured for virtualCluster");
+        }
+        if (listeners.stream().anyMatch(Objects::isNull)) {
+            throw new IllegalConfigurationException("one or more listeners were null");
+        }
+        var names = listeners.stream()
+                .map(VirtualClusterListener::name)
+                .toList();
+        var duplicates = names.stream()
+                .filter(i -> Collections.frequency(names, i) > 1)
+                .collect(Collectors.toSet());
+        if (!duplicates.isEmpty()) {
+            throw new IllegalConfigurationException("The following listener names are duplicated: " + duplicates);
```

Comment:
Think this could be a bit more informative, like "listener names for a virtual cluster must be unique, the following listener names are duplicated".